src/test/java/ru/training/at/hw7jdi/data/DProvider.java

```diff
@@ -0,0 +1,15 @@
+package ru.training.at.hw7jdi.data;
+
+import org.testng.annotations.DataProvider;
+
+public class DProvider {
+
+    @DataProvider(name = "data-provider")
+    public static Object[][] dataProviderMethod() {
+        return new Object[][] {{"JDI_ex8_metalsColorsDataSet.json", 1},
```

Comment:
Data provider should work without any knowledge about the number of rows in source file. 
This was a REQUIREMENT for the task